### PR TITLE
Third Eye and Utsusemi should not stack.

### PR DESCRIPTION
**Pull request by gweivyth**
**_I affirm:_**
- [x] that I agree to Topaz Next's [Limited Contributor License Agreement](https://github.com/topaz-next/topaz/blob/release/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have read the [Contributing Guide](https://github.com/topaz-next/topaz/blob/release/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/topaz-next/topaz/blob/release/CODE_OF_CONDUCT.md)
- [x] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

**_Temporary_**:
- [x] that I understand there are large refactoring efforts going on right now, that these efforts touch every single Lua script and binding, and that my pull request might get put on hold to ensure there are not any conflicts with the refactoring work

image

Above image taken from retail.  These effects shouldn't stack.  Copy Image should wipe Third Eye, and Third Eye should have no effect when player is under the effect of Copy Image.

### CODE REVIEW

**Comment by Xaver-DaRed** [comment]
Tabs need to be converted to 4 spaces/formating, in general.
I would also edit utsusemi_san, becouse theres no reason for it not removing third eye aswell if ichi and ni do.
Sent you a DM to discord with the formated files

**Comment by gweivyth** [comment]
I totally blanked on San even existing cause of 75-brain, added the formatting you sent and the update for San as well, thanks!

File: scripts/globals/spells/utsusemi_ichi.lua [hunk 1]
```diff
@@ -11,8 +11,9 @@ end
 
 function onSpellCast(caster, target, spell)
     local effect = target:getStatusEffect(tpz.effect.COPY_IMAGE)
-
-    -- Get extras shadows
+    and target:delStatusEffect(67) -- Third Eye and Utsusemi don't stack.  Utsusemi removes Third Eye.
```
Comment (ghost) [inline scripts/globals/spells/utsusemi_ichi.lua]: Is there an equivalent status effect for this?  Magic numbers.

File: scripts/globals/spells/utsusemi_ni.lua [hunk 2]
```diff
@@ -11,7 +11,8 @@ end
 
 function onSpellCast(caster, target, spell)
     local effect = target:getStatusEffect(tpz.effect.COPY_IMAGE)
-
+    and target:delStatusEffect(67) -- Third Eye and Utsusemi don't stack.  Utsusemi removes Third Eye.
```
Comment (ghost) [inline scripts/globals/spells/utsusemi_ni.lua]: Same as above, magic numbers.

File: scripts/globals/spells/utsusemi_san.lua [hunk 3]
```diff
@@ -11,6 +11,7 @@ end
 
 function onSpellCast(caster, target, spell)
     local effect = target:getStatusEffect(tpz.effect.COPY_IMAGE)
+    and target:delStatusEffect(67) -- Third Eye and Utsusemi don't stack.  Utsusemi removes Third Eye.
```
Comment (ghost) [inline scripts/globals/spells/utsusemi_san.lua]: Same as above, magic numbers.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by zach2good** [comment]
**EDIT:** Ah, yes, I can read. Ignore everything below. Third eye is the thing being cancelled!

Is checking for `tpz.effect.COPY_IMAGE` enough? I say this because on retail when my gearswap checks for and cancels shadows, I have to check for 4 things: 
```lua
    -- Cancel Shadows
	if spell.english == "Utsusemi: Ichi" then
		if buffactive['Copy Image'] then
			send_command('@wait 2.2; cancel 66')
		elseif buffactive['Copy Image (2)'] then 
			send_command('@wait 2.2; cancel 444')
		elseif buffactive['Copy Image (3)'] then
			send_command('@wait 2.2; cancel 445')
		elseif buffactive['Copy Image (4+)'] then
			send_command('@wait 2.2; cancel 446')
		end
    end
```

File: scripts/globals/spells/utsusemi_ichi.lua (hunk 1 above)
Comment (zach2good) [inline scripts/globals/spells/utsusemi_ichi.lua]: This section will resolve to `local effect = target:getStatusEffect(tpz.effect.COPY_IMAGE) and target:delStatusEffect(67)`, making `effect` a bool (I think). Is that your intention?

Or should the `and` be removed?
```lua
local effect = target:getStatusEffect(tpz.effect.COPY_IMAGE)
target:delStatusEffect(67) -- Third Eye and Utsusemi don't stack.  Utsusemi removes Third Eye.
```

File: scripts/globals/abilities/third_eye.lua [hunk 4]
```diff
@@ -17,5 +17,8 @@ function onAbilityCheck(player, target, ability)
 end
 
 function onUseAbility(player, target, ability)
-    player:addStatusEffect(tpz.effect.THIRD_EYE, 0, 0, 30) --power keeps track of procs
+    if (player:hasStatusEffect(tpz.effect.COPY_IMAGE)) then
+        effect:setMsg(tpz.msg.basic.JA_NO_EFFECT) else          --Returns "no effect" message when Copy Image is active when Third Eye is used.
+        player:addStatusEffect(tpxz.effect.THIRD_EYE, 0, 0, 30) --power keeps track of procs
```
Comment (Xaver-DaRed) [inline scripts/globals/abilities/third_eye.lua]: Typo in this line. tpxz should be changed to tpz.
Also (maybe unrelated and out of the scope of this PR) how does BLINK interact with Third Eye?

**Comment by TeoTwawki** [comment]
> 
> 
> **EDIT:** Ah, yes, I can read. Ignore everything below. Third eye is the thing being cancelled!
> 
> Is checking for `tpz.effect.COPY_IMAGE` enough? I say this because on retail when my gearswap checks for and cancels shadows, I have to check for 4 things:
> 
> ```lua
>     -- Cancel Shadows
> 	if spell.english == "Utsusemi: Ichi" then
> 		if buffactive['Copy Image'] then
> 			send_command('@wait 2.2; cancel 66')
> 		elseif buffactive['Copy Image (2)'] then 
> 			send_command('@wait 2.2; cancel 444')
> 		elseif buffactive['Copy Image (3)'] then
> 			send_command('@wait 2.2; cancel 445')
> 		elseif buffactive['Copy Image (4+)'] then
> 			send_command('@wait 2.2; cancel 446')
> 		end
>     end
> ```

I believe the old ones cheated and used `effect:setIcon()` to use just one effect in reality while sending the correct effectID packets to the client to set the correct icon and tell the client "its multiple effects, honest!" in a similar manner to if we had used `addStatusEffectEx()`.

File: scripts/globals/abilities/third_eye.lua [hunk 5]
```diff
@@ -17,5 +17,9 @@ function onAbilityCheck(player, target, ability)
 end
 
 function onUseAbility(player, target, ability)
-    player:addStatusEffect(tpz.effect.THIRD_EYE, 0, 0, 30) --power keeps track of procs
+    if (player:hasStatusEffect(tpz.effect.COPY_IMAGE)) then
```
Comment (TeoTwawki) [inline scripts/globals/abilities/third_eye.lua:20]: outermost parenthesis are not required. you should use parenthesis on things that are functions like `addStatusEffect()` needs them, or to clarify order of operations like `if a and (b or c) and d then`

So here you would want
```lua
 if player:hasStatusEffect(tpz.effect.COPY_IMAGE) then
```

**Comment by TeoTwawki** [comment]
> Final commit. 

Allegedly! :wink: 

jokes aside safest to be descriptive even on short quick fixes, unless you intend to squash after.

**Comment by zach2good** [comment]
The content is there, but the formatting is off. I'll merge this into a branch and clean it up, then yeet it into release. Thanks for the contribution 👍

**Pull request merged**